packages/components/src/components/hds/app-header/index.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import { tracked } from '@glimmer/tracking';
+import { registerDestructor } from '@ember/destroyable';
+
+export interface HdsAppHeaderSignature {
+  Args: {
+    breakpoint?: string;
+  };
+  Blocks: {
+    logo?: [];
+    globalActions?: [];
+    utilityActions?: [];
+  };
+  Element: HTMLDivElement;
+}
+
+export default class HdsAppHeaderComponent extends Component<HdsAppHeaderSignature> {
+  @tracked isOpen = false;
+  @tracked isDesktop = true;
+  desktopMQ: MediaQueryList;
+
+  desktopMQVal =
+    this.args.breakpoint ||
+    getComputedStyle(document.documentElement).getPropertyValue(
+      '--hds-app-desktop-breakpoint'
+    );
```

Comment:
Consumers can set a width value for `@breakpoint` to optionally override the preset value if they wish. Let me know if this makes sense to do this way... @didoo

Reply:
I think that for the V1 version is OK, but I also think that for the V2 version the whole approach should be revisited: as discussed together, we should try to see if it's possible to move some of the viewport-related logic up to the `AppFrame` level (so it's done once, for both the `AppHeader` and the `SideNav`) and improve the resilience of the `AppHeader` (in terms of responsiveness) using `ResizeObserver`.

The reason why I am saying this is that the way in which we're doing the media querying is based on assumptions about a fixed breaking point (set by us or the consumers) but there are many factors that could make the content overflow or wrap: a different font used by the consumer, an extra long text in the org/project picker, extra content added by the consumers (with potentially variable length) after the org/project picker, etc. All of these variables make my "spider sense" tickle when I think of a fixed value as breaking point between the two modes.

As mentioned during our previous conversation, I don't have an "easy" fix or alternative, so considering the timeline we have, it's OK for the V1 release: it will take some time before the component is actually adopted and used in production, and we can use this time to come up with a more robust approach for the V2 that involves the three different pieces of this puzzle: `AppFrame`, `AppHeader` and `SideNav`.

/cc @alex-ju @Dhaulagiri

Reply:
Ok thank you. I will go ahead and add examples to the Showcase and related tests. Then we can revisit in phase 2 to look towards a more robust solution.

Reply:
I just called the new argument "breakpoint" btw, I don't know if we'd want to come up with some other name.

Reply:
I added a Showcase example and an integration test.